src/python/WMCore/Services/WorkQueue/WorkQueue.py

```diff
@@ -256,6 +256,41 @@ def updatePriority(self, wf, priority):
             wmspec.saveCouch(self.hostWithAuth, self.db.name, dummy_values)
         return
 
+    def updateElementsByWorkflow(self, wf, updateParams, status=None):
+        """
+        Update all available WorkQueue elements of a given workflow  with a set
+        of arguments provided through the `updateParams` dictionary
+        :param wf:           The workflow name
+        :param updateParams: A dictionary with parameters  to be updated
+        :param status:       A list of allowed WorkQueue elements statuses to be considered for updating
+                             Default: None - do not filter by status
+        :return:             No value, raises exceptions from internal methods in case of errors.
+        """
+        # Fetch the whole view with Workqueue elements per given workflow
+        data = self.db.loadView('WorkQueue', 'elementsDetailByWorkflowAndStatus',
+                                {'startkey': [wf], 'endkey': [wf, {}],
+                                 'reduce': False})
+
+        # Fetch only a list of WorkQueue element Ids && Filter them by allowed status
+        if status:
```

Comment:
I like that we are not breaking the current behavior of RequestPriority update.

However, thinking twice about this, if we want to make these updates more efficient, we could update it only for elements in Available, Negotiating and Acquired, from the list of potential statuses for WQE:
https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/WorkQueue/DataStructs/WorkQueueElement.py#L14

If you change it, could you please also update the Issue description (because it says only Available).

Reply:
done